google/cloud/pubsub/internal/publisher_tracing_connection.cc

```diff
@@ -52,14 +115,19 @@ void PublisherTracingConnection::ResumePublish(ResumePublishParams p) {
 }
 
 std::shared_ptr<pubsub::PublisherConnection> MakePublisherTracingConnection(
+    pubsub::Topic topic,
     std::shared_ptr<pubsub::PublisherConnection> connection) {
-  return std::make_shared<PublisherTracingConnection>(std::move(connection));
+  return std::make_shared<PublisherTracingConnection>(std::move(topic),
+                                                      std::move(connection));
 }
 
 #else  // GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY
 
 std::shared_ptr<pubsub::PublisherConnection> MakePublisherTracingConnection(
+    pubsub::Topic topic,
     std::shared_ptr<pubsub::PublisherConnection> connection) {
+  // Silence the unused parameter warnings.
+  (void)topic;
```

Comment:
I believe you can just not give the parameter a name to achieve the same effect.